core/controllers/android.py

```diff
@@ -231,6 +240,33 @@ def get(self) -> None:
                         subtopic_page.to_dict() if subtopic_page is not None
                         else None)
                 })
+        elif activity_type == 'question':
+            for activity_data in activities_data:
+                question = question_fetchers.get_question_by_id(
+                    [activity_data['id']],
+                    strict=False,
+                    version=activity_data.get('version'))
+                activities.append({
+                    'id': activity_data['id'],
+                    'version': activity_data.get('version'),
+                    'payload': (
+                        question.to_dict() if question is not None else None)
+                })
+        elif activity_type == 'question_skill_link':
+            for activity_data in activities_data:
+                question_skill_link_models = (
+                    question_models.QuestionSkillLinkModel
```

Comment:
This shouldn't directly reference the storage layer, yeah. 

Re load, thoughts on using question_fetchers.get_questions_and_skill_descriptions_by_skill_ids() and passing in multiple skill IDs at a time? Otherwise maybe you'll need to make a new function in question_fetchers for this use case.

Reply:
Yeah, the latter is a good idea and what I've implemented in the latest version. I also introduced new ``question_fetchers`` functionality to avoid the direct storage model reference here.